src/ophys_etl/transforms/data_loaders.py

```diff
@@ -0,0 +1,63 @@
+import pandas as pd
+import numpy as np
+from collections import namedtuple
+
+motion_border = namedtuple('motion_border', 'left right up down')
+
+
+def get_max_correction_values(motion_corr_df: pd.DataFrame,
+                              max_shift: float = 30.0) -> motion_border:
+    """
+    Gets the max correction values in the cardinal directions from
+    a motion correction dataframe created during Motion Correction workflow
+    step.
+    Parameters
+    ----------
+    motion_corr_df: pd.DataFrame
+        The dataframe that contains the motion correction value calculated
+        during motion correction workflow step of ophys segmentation pipeline.
+    max_shift: float
+        Maximum shift to allow when considering motion correction. Any
+        larger shifts are considered outliers.
+
+    Returns
+    -------
+    motion_border
+        A named tuple containing the maximum correction values found during
+        motion correction workflow step. Saved with the direction is the order
+        [Left, Right, Up, Down] and also with names.
+
+    """
+    # validate if input columns match supported versions of motion correction
+    # files
+    input_column_set = set(motion_corr_df.columns)
+    required_columns = {'x', 'y'}
+    if not required_columns.intersection(input_column_set) == required_columns:
+        raise KeyError("Required columns to compute max correction not in "
+                       f"input dataset. Supplied columns: {input_column_set}, "
+                       f"need columns {required_columns}")
+    if max_shift <= 0:
+        raise ValueError(f"Max Shift input: {max_shift}. A positive input"
+                         f"input for max shift is required.")
+
+    # filter based out analomies based on maximum_shift
+    x_no_outliers = motion_corr_df["x"][(motion_corr_df["x"] >= -max_shift) &
+                                        (motion_corr_df["x"] <= max_shift)]
+    y_no_outliers = motion_corr_df["y"][(motion_corr_df["y"] >= -max_shift) &
+                                        (motion_corr_df["y"] <= max_shift)]
+    # calculate max border shifts
+    right_shift = -1 * x_no_outliers.min()
```

Comment:
Why did you change these lines from the AllenSDK? The SDK ensures that there are no negative values (will be 0 instead).

```
    right_shift = np.max(-1*x_no_outliers.min(), 0)
    down_shift = np.max(-1*y_no_outliers.min(), 0)
```

Reply:
I assumed negative values were still relevant, I can add these back in.

Reply:
If we're porting stuff from the SDK in order to not use it, we need to duplicate the behavior.

Reply:
They've been added back in, the logic is now the same

Reply:
I actually remember why I got rid of this originally, the np.max is not restricting to over 0 it is taking the max along axis 0 of the input, which is already a single number. That is why I got rid of it as it served no purpose IMO but I've added it back in so that the two methods look identical.